File: octgnFX/agsxmpp/trunk/samples/VS2008/MiniClient/frmAddRoster.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

using agsXMPP;

namespace MiniClient
{
	/// <summary>
	/// Summary description for femAddRoster.
	/// </summary>
	public class frmAddRoster : System.Windows.Forms.Form
	{
		private System.Windows.Forms.TextBox txtNickname;
		private System.Windows.Forms.TextBox txtJid;
		private System.Windows.Forms.Label label3;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Button cmdAdd;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		private XmppClientConnection _connection;
        
		public frmAddRoster(XmppClientConnection con)
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			_connection = con;
		}

        public frmAddRoster(Jid jid, XmppClientConnection con)
            : this(con)
        {
            txtJid.Text = jid.Bare;
        }

        public frmAddRoster(Jid jid, string nickname, XmppClientConnection con)
            : this(jid, con)
        {
            txtNickname.Text = nickname;
        }

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.txtNickname = new System.Windows.Forms.TextBox();
            this.txtJid = new System.Windows.Forms.TextBox();
            this.label3 = new System.Windows.Forms.Label();
            this.label2 = new System.Windows.Forms.Label();
            this.label1 = new System.Windows.Forms.Label();
            this.cmdAdd = new System.Windows.Forms.Button();
            this.SuspendLayout();
            // 
            // txtNickname
            // 
            this.txtNickname.Location = new System.Drawing.Point(80, 56);
            this.txtNickname.Name = "txtNickname";
            this.txtNickname.Size = new System.Drawing.Size(168, 20);
            this.txtNickname.TabIndex = 9;
            // 
            // txtJid
            // 
            this.txtJid.Location = new System.Drawing.Point(80, 8);
            this.txtJid.Name = "txtJid";
            this.txtJid.Size = new System.Drawing.Size(168, 20);
            this.txtJid.TabIndex = 8;
            // 
            // label3
            // 
            this.label3.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label3.Location = new System.Drawing.Point(8, 56);
            this.label3.Name = "label3";
            this.label3.Size = new System.Drawing.Size(64, 16);
            this.label3.TabIndex = 11;
            this.label3.Text = "Nickname:";
            // 
            // label2
            // 
            this.label2.Location = new System.Drawing.Point(80, 32);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(160, 16);
            this.label2.TabIndex = 10;
            this.label2.Text = "user@server.org";
            // 
            // label1
            // 
            this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label1.Location = new System.Drawing.Point(8, 8);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(72, 16);
            this.label1.TabIndex = 7;
            this.label1.Text = "Jabber ID:";
            // 
            // cmdAdd
            // 
            this.cmdAdd.FlatStyle = System.Windows.Forms.FlatStyle.System;
            this.cmdAdd.Location = new System.Drawing.Point(96, 88);
            this.cmdAdd.Name = "cmdAdd";
            this.cmdAdd.Size = new System.Drawing.Size(80, 24);
            this.cmdAdd.TabIndex = 12;
            this.cmdAdd.Text = "Add";
            this.cmdAdd.Click += new System.EventHandler(this.cmdAdd_Click);
            // 
            // frmAddRoster
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(264, 118);
            this.Controls.Add(this.cmdAdd);
            this.Controls.Add(this.txtNickname);
            this.Controls.Add(this.txtJid);
            this.Controls.Add(this.label3);
            this.Controls.Add(this.label2);
            this.Controls.Add(this.label1);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
            this.MaximizeBox = false;
            this.MinimizeBox = false;
            this.Name = "frmAddRoster";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "Add Contact";
            this.ResumeLayout(false);
            this.PerformLayout();

		}
		#endregion

		private void cmdAdd_Click(object sender, System.EventArgs e)
		{
			Jid jid = new Jid(txtJid.Text);

			// Add the Rosteritem using the Rostermanager			
			if (txtNickname.Text.Length > 0)
				_connection.RosterManager.AddRosterItem(jid, txtNickname.Text);				
			else
				_connection.RosterManager.AddRosterItem(jid);

			// Ask for subscription now
			_connection.PresenceManager.Subscribe(jid);
						
			this.Close();
		}
	}
}

File: octgnFX/agsxmpp/trunk/samples/VS2003/MiniClientVB/frmAddRoster.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Imports agsXMPP

Namespace MiniClient

	Public Class frmAddRoster
	Inherits System.Windows.Forms.Form
		Private components As System.ComponentModel.Container = Nothing
		Private _connection As XmppClientConnection
		Private label2 As System.Windows.Forms.Label
		Private txtNickname As System.Windows.Forms.TextBox
		Private txtJid As System.Windows.Forms.TextBox
		Private cmdAdd As System.Windows.Forms.Button
		Private label1 As System.Windows.Forms.Label
		Private label3 As System.Windows.Forms.Label
		
		Public Sub New(ByVal con As XmppClientConnection)
			InitializeComponent
			_connection = con
		End Sub

		Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not (components Is Nothing) Then
					components.Dispose
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		Private Sub InitializeComponent()
			Me.label3 = New System.Windows.Forms.Label
			Me.label1 = New System.Windows.Forms.Label
			Me.cmdAdd = New System.Windows.Forms.Button
			Me.txtJid = New System.Windows.Forms.TextBox
			Me.txtNickname = New System.Windows.Forms.TextBox
			Me.label2 = New System.Windows.Forms.Label
			Me.SuspendLayout
			'
			'label3
			'
			Me.label3.Location = New System.Drawing.Point(8, 56)
			Me.label3.Name = "label3"
			Me.label3.Size = New System.Drawing.Size(64, 16)
			Me.label3.TabIndex = 11
			Me.label3.Text = "Nickname:"
			'
			'label1
			'
			Me.label1.Location = New System.Drawing.Point(8, 8)
			Me.label1.Name = "label1"
			Me.label1.Size = New System.Drawing.Size(72, 16)
			Me.label1.TabIndex = 7
			Me.label1.Text = "Jabber ID:"
			'
			'cmdAdd
			'
			Me.cmdAdd.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.cmdAdd.Location = New System.Drawing.Point(96, 88)
			Me.cmdAdd.Name = "cmdAdd"
			Me.cmdAdd.Size = New System.Drawing.Size(80, 24)
			Me.cmdAdd.TabIndex = 12
			Me.cmdAdd.Text = "Add"
			AddHandler Me.cmdAdd.Click, AddressOf Me.cmdAdd_Click
			'
			'txtJid
			'
			Me.txtJid.Location = New System.Drawing.Point(80, 8)
			Me.txtJid.Name = "txtJid"
			Me.txtJid.Size = New System.Drawing.Size(168, 20)
			Me.txtJid.TabIndex = 8
			Me.txtJid.Text = ""
			'
			'txtNickname
			'
			Me.txtNickname.Location = New System.Drawing.Point(80, 56)
			Me.txtNickname.Name = "txtNickname"
			Me.txtNickname.Size = New System.Drawing.Size(168, 20)
			Me.txtNickname.TabIndex = 9
			Me.txtNickname.Text = ""
			'
			'label2
			'
			Me.label2.Location = New System.Drawing.Point(80, 32)
			Me.label2.Name = "label2"
			Me.label2.Size = New System.Drawing.Size(160, 16)
			Me.label2.TabIndex = 10
			Me.label2.Text = "user@server.org"
			'
			'frmAddRoster
			'
			Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
			Me.ClientSize = New System.Drawing.Size(264, 118)
			Me.Controls.Add(Me.cmdAdd)
			Me.Controls.Add(Me.txtNickname)
			Me.Controls.Add(Me.txtJid)
			Me.Controls.Add(Me.label3)
			Me.Controls.Add(Me.label2)
			Me.Controls.Add(Me.label1)
			Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
			Me.MaximizeBox = false
			Me.MinimizeBox = false
			Me.Name = "frmAddRoster"
			Me.Text = "Add Contact"
			Me.ResumeLayout(false)
		End Sub

		Private Sub cmdAdd_Click(ByVal sender As Object, ByVal e As System.EventArgs)
			Dim jid As Jid = New Jid(txtJid.Text)
			If txtNickname.Text.Length > 0 Then
				_connection.RosterManager.AddRosterItem(jid, txtNickname.Text)
			Else
				_connection.RosterManager.AddRosterItem(jid)
			End If
			_connection.PresenceManager.Subcribe(jid)
			Me.Close
		End Sub
	End Class 
End Namespace
